### DISCUSSION TITLE
Catcher flips on argon skin

### DISCUSSION THREAD

**Discussion by GamingWithJanko** [ppy/osu#22690] [Q&A] [answered]
### Type

Game behaviour

### Bug description

The CTB catcher in Lazer (Skin Argon) flips in gameplay (Using RX)

### Screenshots or videos

File was too big... https://youtu.be/WKCfH36E3rM

0:00 Trying to finds a note to jiggle around.
0:45 Trying to actually play.

I'm sorry about the Waveform...
Forgot to disable it. :/

### Version

Lazer 2023.207.0

### Logs

Not needed, because it's a GRAPHICAL BUG.

**Comment by bdach** [comment]
Migrating to discussion as this is a subjective issue.

I was tangentially aware of this but I don't strongly feel that this is a bug. Maybe it shouldn't flip on argon (because that's a skin that has a completely geometrical catcher) but this probably needs discussion first.

**Reply by Axilotl17** [reply]
I don't play taiko, but just looking at the current behavior I can see why this isn't appreciated. 

the way it flips back and fourth when it lands on the edge does look really distracting.

https://user-images.githubusercontent.com/78441708/219981391-fc7ed183-a5cb-4a59-826a-ae8a8dcbfe8e.mp4

**Reply by GamingWithJanko** [reply]
What @Axilotl17 said.

**Reply by GamingWithJanko** [reply]
@bdach you might wanna rethink on that move decision.

It prevents proper gameplay, as it really is distracting.

Please put the topic back where it belongs.

**Reply by bdach** [reply]
Let me counter with a question then: why is this about argon specifically, if this is the behaviour that all skins have, including legacy skins with similarly geometrical catchers?

**Reply by Axilotl17** [reply]
I was thinking the same thing. some skins should totally flip the notes, like the old catch skin, however geometric ones shouldn't. maybe this should be a setting, either in lazer or for skins? Lazer skinning could totally have this although enough of these one-off solutions and toggles could cause the skinning system to become bloated.

also @GamingWithJanko this is clearly not an issue, sorry. I agree with bdach that this should be a discussion.

**Reply by GamingWithJanko** [reply]
The catcher should flip, yes.

But the notes on the catcher... I never saw these flipping before.

I can literally only read Argon.

For the rest, I don't know, but the notes shouldn't flip with the catcher.

If that's intentional, then remove that discussion. If it isn't, it's something that needs patching.

**Reply by Axilotl17** [reply]
>The catcher should flip, yes.
But the notes on the catcher... I never saw these flipping before.

I think it should be both or neither. it doesn't make sense to flip the catcher and not the notes. 

>I can literally only read Argon.
For the rest, I don't know, but the notes shouldn't flip with the catcher.

what?

>If that's intentional, then remove that discussion. If it isn't, it's something that needs patching.

the discussion is so we can decide how to handle this moving forward. it isn't necessarily a patch, but a change. the current status is intentional, but maybe it should be redone.

**Reply by mk56-spn** [reply]
To build on what axolotl  said, it being changed to a discussion shouldn't be seen as a discarding, issues are *usually* used for when direction is clear to the devs and it's just a question of getting it done at some point. Whereas here direction is yet to be established. So instead of crowding an issue a discussion is made so people can voice their opinions and hopefully reach a consensus with the Devs.

**Reply by GamingWithJanko** [reply]
Fair enough, @mk56-spn

**Comment by Axilotl17** [comment]
Okay, so, catcher direction:

in some instances, like the argon skin, the catcher is **symmetrical** (geometric doesnt mean symmetrical) and so flipping it doesn't change anything visually. however, the notes that have landed on the catcher are still present and not symmetrical, creating a weird flip visual which I think we can all agree is distracting. 

however, if the catcher flips regardless, like the triangles skin or most other legacy skins, then the notes flipping with the catcher actually makes sense. if they don't flip, it creates a disconnect and looks really choppy.

i think there are 4 real options here; 
1. there could be a skin toggle/setting for skinners to change how they will.
2. argon could be hard coded specially (although this would be annoying for skinners).
3. it could be a toggleable setting for users to change based on what they prefer.
4. its a feature!! a really bad looking feature, but any solution would be a bad solution.

**Reply by GamingWithJanko** [reply]
In that case, I would go for 1 and 3

1 is set by the skinner.
But 3 allows the user to override it.

**Reply by Axilotl17** [reply]
good idea. i concur.

**Comment by rrex971** [comment]
Adding on to this discussion: I've never seen any legacy skins flip the fruits with the catcher. Stable flips the catcher itself but not the fruits on it and there's nothing in `skin.ini` that can allow this behavior with skins at all. Perhaps it could be added as an option but the default behavior shouldn't flip the fruits along with the catcher as it can get quite distracting, regardless of whether the skin has a symmetrical catcher or not. 

Edit: oops, just realized this behavior is only with argon/classic and legacy skins work as they should (#13747). didn't realize this and accidentally dug up an old thread

**Reply by GamingWithJanko** [reply]
Hey, it happens. XD

**Comment by bdach** [comment, marked as answer]
will be turned off entirely by https://github.com/ppy/osu/pull/27165

**Reply by GamingWithJanko** [reply]
Sweet.
